subsys/net/lib/lwm2m/lwm2m_obj_server.c

```diff
@@ -90,6 +104,55 @@ static int update_trigger_cb(u16_t obj_inst_id)
 #endif
 }
 
+void *binding_pre_write(u16_t obj_inst_id, size_t *data_len)
+{
+	*data_len = TRANSPORT_BINDING_LEN;
+	return binding_mode_validation_buf;
+}
+
+static int binding_validate(u16_t obj_inst_id, u8_t *data, u16_t data_len,
+			    bool last_block, size_t total_size)
+{
+	u16_t mode, len;
+
+	for (mode = BINDING_MODE_U; mode < BINDING_MODE_UNKNOWN; mode++) {
+		len = strlen(binding_modes[mode]);
+
+		if (len == data_len && !strcmp(binding_modes[mode], data)) {
+			break;
+		}
+	}
+
+	/* only UDP mode is supported for now */
+	if (mode != BINDING_MODE_U || mode == BINDING_MODE_UNKNOWN) {
+		SYS_LOG_ERR("Unsupported binding mode: %s",
+			    mode == BINDING_MODE_UNKNOWN ?
+			    "unknown" : binding_modes[mode]);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int binding_post_write(u16_t obj_inst_id, u8_t *data, u16_t data_len,
```

Comment:
Add a comment here explaining that because of the alternate buffer for validation this callback is needed to place the data into the regular storage buffer (post validation).